File: javaorders/src/main/java/tech/gebel/javaorders/models/Agents.java
package tech.gebel.javaorders.models;

import javax.persistence.*;

@Entity
@Table(name = "agents")
public class Agents {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    @Column(name = "agentcode")
    private long agentCode;

    @Column(name = "agentname")
    private String agentName;

    private double commission;
    private String country;
    private String phone;

    @Column(name = "workingarea")
    private String workingArea;

    public Agents() {
    }

    public Agents(String agentName, double commission, String country, String phone, String workingArea) {
        this.agentName = agentName;
        this.commission = commission;
        this.country = country;
        this.phone = phone;
        this.workingArea = workingArea;
    }
}

File: src/main/kotlin/com/wkdrabbit/crudycustomers/model/Agents.kt
package com.wkdrabbit.crudycustomers.model

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import org.graalvm.compiler.core.common.type.ArithmeticOpTable
import sun.util.resources.Bundles
import javax.persistence.*

@Entity
@Table(name = "agent")
class Agents{

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    var agentId : Long = 0

    @Column(unique = true, nullable = false)
    var name : String = ""


    var agentName = ""
    var workingArea = ""
    var commission = 0.0
    var phone = ""
    var country = ""

    @OneToMany(mappedBy = "agent",
            cascade = [CascadeType.ALL],
            orphanRemoval = true)
    @JsonIgnoreProperties("agent")
    val customers = mutableListOf<Customers>()


    @OneToMany(mappedBy = "agent",
            cascade = [CascadeType.ALL],
            orphanRemoval = true)
    @JsonIgnoreProperties("agent")
    val orders = arrayListOf<Orders>()


    constructor(agentName: String, workingArea: String, commission: Double, phone: String, country: String) {
        this.agentName = agentName
        this.workingArea = workingArea
        this.commission = commission
        this.phone = phone
        this.country = country
    }

    constructor()


}